src/lib.rs

```diff
@@ -337,7 +332,7 @@ where
         &self,
         key: StorageKey,
         at_block: Option<Hash>,
-    ) -> Option<V> {
+    ) -> WsResult<V> {
```

Comment:
this is problematic. There are basically two things that can go wrong: communication and storage lookup.
A WsResult should indicate a communication error but we also need the `Option` to indicate if the storage key has yielded an entry or not. so, probably, this should actually be `WsResult<Option<V>>`